### PR TITLE
DAT-4113 Nav redesign

### PR DESCRIPTION
**Pull request by rybmat**
## Links

* https://wikia-inc.atlassian.net/browse/DAT-4113

## Description

* adding search icon on site-head bar, 
* separating navigation and search,
* .

## Reviewers

@Wikia/west-wing

### CODE REVIEW

File: front/main/app/components/wikia-nav.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+export default Ember.Component.extend({
+	defaultNavContent: 'side-nav-global-navigation-root',
+
+	init() {
+		this._super(...arguments);
+
+		this.currentNavContent = this.get('defaultNavContent');
+	},
+
+	willDestroyElement() {
```
Comment (RafLeszczynski) [inline front/main/app/components/wikia-nav.js]: I'm not a big fan of this approach but it's required if we want to keep replacing nav content the way it's done in:
https://github.com/Wikia/mercury/pull/2402/files#diff-48248ead84b56ab3415af1dcda3499eeR36

any ideas ?

File: front/main/app/components/wikia-nav.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+		/**
+		 * @returns {void}
+		 */
+		loadRandomArticle() {
+			this.set('globalNavContent', 'side-nav-global-navigation-root');
+			this.sendAction('loadRandomArticle');
+		},
+
+		replaceNavigationContent(navName) {
```
Comment (RafLeszczynski) [inline front/main/app/components/wikia-nav.js]: consider changing to switch statment

File: front/main/app/components/wikia-nav.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+		 */
+		collapse() {
+			this.set('globalNavContent', 'side-nav-global-navigation-root');
+			this.sendAction('toggleVisibility', false);
+		},
+
+		/**
+		 * @returns {void}
+		 */
+		loadRandomArticle() {
```
Comment (RafLeszczynski) [inline front/main/app/components/wikia-nav.js]: don't need it anymore as reset to default nav state is already done on `willDestroyElement`. Could just use `loadRandomArticle` from parent as `this.attrs. loadRandomArticle` in hbs template

File: front/main/app/components/wikia-search.js [hunk 4]
```diff
@@ -57,15 +57,6 @@ export default Ember.Component.extend(
 				window.location.assign(`${Mercury.wiki.articlePath}Special:Search?search=${value}&fulltext=Search`);
 			},
 
-			searchFocus() {
```
Comment (RafLeszczynski) [inline front/main/app/components/wikia-search.js]: we would probably want auto focus when search is open

Comment (RafLeszczynski) [inline front/main/app/components/wikia-search.js]: could try using default handlebars autofocus in template

File: front/main/app/routes/application.js [hunk 5]
```diff
@@ -212,7 +212,7 @@ export default Route.extend(
 			 * @returns {void}
 			 */
 			toggleSideNav(visible) {
```
Comment (RafLeszczynski) [inline front/main/app/routes/application.js]: it seams that there are some additional uses of `toggleSideNav` on route. we should rename it to `toggleDrawer` to be consistent. also check this place and fix it: https://github.com/Wikia/mercury/pull/2402/files#diff-f25b80f4a8250372111d3cb2b7567752R154

File: front/main/app/styles/component/_site-head.scss [hunk 6]
```diff
@@ -4,29 +4,52 @@ $z-header: $z-6;
...
-	height: rem-calc($site-head-height);
+	display: flex;
+	flex-direction: row;
+	flex-wrap: nowrap;
+	height: $site-head-height;
+	justify-content: space-between;
 	left: 0;
 	margin: 0 auto;
 	position: fixed;
+	padding-left: 4px;
```
Comment (RafLeszczynski) [inline front/main/app/styles/component/_site-head.scss]: could be moved to local variable

File: front/main/app/styles/component/_site-head.scss [hunk 7]
```diff
@@ -4,29 +4,52 @@ $z-header: $z-6;
...
+	padding-right: 4px;
 	right: 0;
 	text-align: center;
 	top: 0;
 	z-index: $z-site-head;
 
+	&.ember-view {
```
Comment (RafLeszczynski) [inline front/main/app/styles/component/_site-head.scss]: please check where this selectors are added to site-head and see if they are still needed. If not remove and remove this code. If they are needed add appropriate comment here why we are overwriting them with `display:none`.

File: front/main/app/styles/component/_site-head.scss [hunk 8]
```diff
@@ -36,6 +59,15 @@ $site-head-height: 50; //unitless
 		}
 	}
 
+	.nav-button {
+		position: relative;
+		svg {
+			height: 24px;
+			width: 25px;
+			fill: #39414d;
```
Comment (RafLeszczynski) [inline front/main/app/styles/component/_site-head.scss]: do not use hardcoded color values

File: front/main/app/styles/component/_site-head.scss [hunk 9]
```diff
@@ -36,6 +59,15 @@ $site-head-height: 50; //unitless
 		}
 	}
 
+	.nav-button {
```
Comment (RafLeszczynski) [inline front/main/app/styles/component/_site-head.scss]: is this style for a blue dot or something else? we should be more specific with the SVG style declaration here

File: front/main/app/templates/components/site-head.hbs [hunk 10]
```diff
@@ -1,11 +1,22 @@
...
 		{{svg 'wikia-logo' role='img' class='icon wikia-logo'}}
 	</a>
 </div>
+
+{{#wikia-ui-components/icon-button
+	click=(action 'siteHeadIconClick' navIcon)
+	class='nav-button site-head-icon-nav'
+	icon=visibleNavIcon
+}}
+	<!--TODO: remove this svg when spitfires remove blueDot functionality-->
```
Comment (RafLeszczynski) [inline front/main/app/templates/components/site-head.hbs]: please use handlebars comment

File: front/main/app/templates/components/wikia-drawer.hbs [hunk 11]
```diff
@@ -0,0 +1,10 @@
+<div class='side-nav-menu'>
```
Comment (RafLeszczynski) [inline front/main/app/templates/components/wikia-drawer.hbs]: check if this additional inner wrapper is needed

File: front/main/app/templates/components/wikia-nav.hbs [hunk 12]
```diff
@@ -0,0 +1,7 @@
+{{component
+	currentNavContent
+	class='side-nav-menu__contents'
+	replaceNavigationContent=(action 'replaceNavigationContent')
+	collapse=(action 'collapse')
+	loadRandomArticle=(action 'loadRandomArticle')
+}}
```
Comment (RafLeszczynski) [inline front/main/app/templates/components/wikia-nav.hbs]: missing new line

File: front/main/app/templates/components/wikia-ui-components/icon-button.hbs [hunk 13]
```diff
@@ -1 +1,2 @@
-{{svg icon role='img' class='icon-button-icon' width=iconSize height=iconSize}}
\ No newline at end of file
+{{svg icon role='img' class='icon-button-icon' width=iconSize height=iconSize}}
+{{yield}}
```
Comment (RafLeszczynski) [inline front/main/app/templates/components/wikia-ui-components/icon-button.hbs]: if this is only for the blue dot and it's gone be removed next week by spitfires we should rething if we need this addtional functionality in icon button

File: front/main/app/components/site-head.js [hunk 14]
```diff
@@ -2,48 +2,84 @@ import Ember from 'ember';
...
+		isNewBadgeVisible: computed.alias('shouldDisplayNewBadge'),
+
+		// TODO: Remove this code after spitfires remove BlueDot functionality
+		shouldDisplayNewBadge: computed('newFeaturesBadges.features.[]', 'visibleNavIcon', function () {
+			return this.get('newFeaturesBadges').shouldDisplay('recent-wiki-activity') &&
+				this.get('visibleNavIcon') !== this.get('closeIcon');
+		}),
+
+		visibleNavIcon: computed('activeIcon', function () {
+			return this.get('activeIcon') !== this.get('navIcon') ? this.get('navIcon') : this.get('closeIcon');
```
Comment (dianafa) [inline front/main/app/components/site-head.js]: can we cache `this.get('navIcon')` and turn this condition to be `===` ?

File: front/main/app/components/site-head.js [hunk 15]
```diff
@@ -2,48 +2,84 @@ import Ember from 'ember';
...
+		}),
+
+		visibleSearchIcon: computed('activeIcon', function () {
+			return this.get('activeIcon') !== this.get('searchIcon') ? this.get('searchIcon') : this.get('closeIcon');
```
Comment (dianafa) [inline front/main/app/components/site-head.js]: can we cache this.get('searchIcon') and turn this condition to be === ?

File: front/main/app/components/site-head.js (hunk 14 above)
Comment (dianafa) [inline front/main/app/components/site-head.js]: also, I'd change name of this property - it currently seems for me like it returns a bool value, but it returns string - name of icon instead

File: front/main/app/styles/component/_site-head.scss (hunk 7 above)
Comment (rybmat) [inline front/main/app/styles/component/_site-head.scss]: Tried to find where it is added but I couldn't find anything.

File: front/main/app/components/site-head.js [hunk 16]
```diff
@@ -2,48 +2,84 @@ import Ember from 'ember';
...
+				} else {
+					track({
+						action: trackActions.click,
+						category: 'side-nav',
+						label: `${icon}-collapsed`
+					});
+
+					this.sendAction('toggleDrawer', false);
+					this.get('setDrawerContent')(null);
+				}
```
Comment (dianafa) [inline front/main/app/components/site-head.js]: this function is not very nice but I assume it's gonna be cleaner after BlueDot removal

File: front/main/app/styles/component/_site-head.scss (hunk 9 above)
Comment (rybmat) [inline front/main/app/styles/component/_site-head.scss]: for both buttons within nav bar

File: front/main/app/components/wikia-nav.js (hunk 1 above)
Comment (dianafa) [inline front/main/app/components/wikia-nav.js]: after talk with @idradm - he'll probably be removing this check you're referring to so this won't be needed

File: front/main/app/templates/components/wikia-ui-components/icon-button.hbs (hunk 13 above)
Comment (dianafa) [inline front/main/app/templates/components/wikia-ui-components/icon-button.hbs]: can't we just add note to remove it like in other places in our code?\

File: front/main/app/components/application-wrapper.js [hunk 17]
```diff
@@ -25,18 +27,22 @@ import {trackPerf} from 'common/utils/track-perf';
...
 	classNames: ['application-wrapper'],
 	classNameBindings: ['smartBannerVisible', 'verticalClass'],
+	drawerContentNavigation: 'nav',
+	drawerContentSearch: 'search',
+	activeDrawerContent: null,
+
+	wikiaHomepage: Ember.getWithDefault(Mercury, 'wiki.homepage', 'http://www.wikia.com'),
+
+	drawerContentComponent: computed('activeDrawerContent', function () {
+		return `wikia-${this.get('activeDrawerContent')}`;
```
Comment (RafLeszczynski) [inline front/main/app/components/application-wrapper.js]: we could move `wikia-` prefix to separate const to improve readability

File: front/main/app/components/site-head.js (hunk 16 above)
Comment (RafLeszczynski) [inline front/main/app/components/site-head.js]: I don't like the additional condition for blue dot tracking. @dianafa any ideas how to improve?

File: front/main/app/components/wikia-nav.js (hunk 1 above)
Comment (RafLeszczynski) [inline front/main/app/components/wikia-nav.js]: ok lets leave it for next iteration

File: front/main/app/styles/component/_site-head.scss (hunk 7 above)
Comment (RafLeszczynski) [inline front/main/app/styles/component/_site-head.scss]: it cannot came from nowhere :-)

File: front/main/app/components/application-wrapper.js [hunk 18]
```diff
@@ -1,6 +1,8 @@
 import Ember from 'ember';
 import {trackPerf} from 'common/utils/track-perf';
 
+const {computed} = Ember;
```
Comment (rogatty) [inline front/main/app/components/application-wrapper.js]: If you do that, why not destructure all `Ember.*`?

File: front/main/app/components/application-wrapper.js [hunk 19]
```diff
@@ -25,18 +27,22 @@ import {trackPerf} from 'common/utils/track-perf';
...
+	verticalClass: computed(() => {
 		const vertical = Ember.get(Mercury, 'wiki.vertical');
 
 		return `${vertical}-vertical`;
 	}),
 
-	noScroll: false,
-	scrollLocation: null,
-	smartBannerVisible: false,
-	firstRender: true,
```
Comment (rogatty) [inline front/main/app/components/application-wrapper.js]: This shouldn't be removed.

Comment (rogatty) [inline front/main/app/components/application-wrapper.js]: Actually, I don't think any of these 4 should be removed as they're all used.

File: front/main/app/components/site-head.js [hunk 20]
```diff
@@ -1,44 +1,94 @@
 import Ember from 'ember';
 import {track, trackActions} from 'common/utils/track';
 
+const {computed, inject} = Ember;
```
Comment (rogatty) [inline front/main/app/components/site-head.js]: Why not `Component` too?

File: front/main/app/components/site-head.js [hunk 21]
```diff
@@ -1,44 +1,94 @@
...
-	newFeaturesBadges: Ember.inject.service(),
-	shouldDisplayNewBadge: Ember.computed('newFeaturesBadges.features.[]', function () {
-		return this.get('newFeaturesBadges').shouldDisplay('recent-wiki-activity');
+		return this.get('activeIcon') === navIcon ? this.get('closeIcon') : navIcon;
 	}),
 
+	visibleSearchIconName: computed('activeIcon', function () {
+		const searchIcon = this.get('searchIcon');
+
+		return this.get('activeIcon') === searchIcon ? this.get('closeIcon') : searchIcon;
```
Comment (rogatty) [inline front/main/app/components/site-head.js]: `visibleSearchIconName` can return `'close'` - this is some twisted logic or bad naming ;) Can it be done in some other, more clean way?

File: front/main/app/components/site-head.js [hunk 22]
```diff
@@ -1,44 +1,94 @@
...
-			this.sendAction('toggleSideNav', true);
-		},
 
-		/**
-		 * @returns {void}
-		 */
-		showUserMenu() {
-			this.sendAction('toggleUserMenu', true);
+				this.get('setDrawerContent')(null);
+				this.sendAction('toggleDrawer', false);
```
Comment (rogatty) [inline front/main/app/components/site-head.js]: Can we make it `this.get('toggleDrawer')(false)` for consistency?

File: front/main/app/components/site-head.js (hunk 21 above)
Comment (rybmat) [inline front/main/app/components/site-head.js]: 'close' is a name of svg file with icon, so do you suggest to rename it?

File: front/main/app/styles/component/_site-head.scss (hunk 7 above)
Comment (rybmat) [inline front/main/app/styles/component/_site-head.scss]: don't know what happened, but now it is working correctly without this piece of code :p

File: front/main/app/components/application-wrapper.js (hunk 19 above)
Comment (rybmat) [inline front/main/app/components/application-wrapper.js]: done

**Pull request merged**